crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -107,6 +107,34 @@ impl Analyzer<'_, '_> {
             | Type::Param(_)
             | Type::Module(_)
             | Type::Tpl(..) => return Ok(ty),
+            Type::Interface(Interface {
+                span,
+                name,
+                type_params,
+                extends,
+                body,
+                metadata: InterfaceMetadata { common },
+            }) => {
+                for extend in extends {
+                    let types = extend.to_owned().type_args.into_iter().flat_map(|cur| cur.params);
+                    let union_type = Type::new_union(*span, types);
```

Comment:
It almost solved nest_assignment errors but not RegExpExecArray yet.
Could you give me any hint where to get RegExpExecArray type to return Array?

Reply:
`RegExpExecArray` is declared as

```ts
interface RegExpExecArray extends Array<string> {
    index: number;
    input: string;
}
```

So you need to check the type arguments of `Array`